www/caddy/src/opnsense/mvc/app/models/OPNsense/Caddy/Caddy.php

```diff
@@ -107,6 +108,52 @@ private function checkSubdomainsAgainstDomains($subdomains, $domains, $messages)
         }
     }
 
+    // 4. Get the current OPNsense WebGUI ports
+    private function getWebGuiPorts() {
+        $webgui = Config::getInstance()->object()->system->webgui ?? null;
+        $webGuiPorts = [];
+
+        // Only add ports to array if no specific interfaces for the WebGUI are set
+        if (!empty($webgui) && empty((string)$webgui->interfaces)) {
+            // Add port 443 if no specific port is set, otherwise set custom webgui port
+            if (empty($webgui->port)) {
+                $webGuiPorts[] = !empty((string)$webgui->port) ? (string)$webgui->port : '443';
+            }
+
+            // Add port 80 if HTTP redirect is not explicitly disabled
+            if (empty((string)$webgui->disablehttpredirect)) {
+                $webGuiPorts[] = '80';
+            }
+        }
+
+        return $webGuiPorts;
+    }
+
+    // 4. Check for conflicts between Caddy and OPNsense WebGUI ports
+    private function checkWebGuiSettings($messages) {
+        $overlap = array_intersect($this->getWebGuiPorts(), ['80', '443']);
+        $tlsAutoHttpsSetting = (string)$this->general->TlsAutoHttps;
+
+        if (!empty($overlap) && $tlsAutoHttpsSetting !== 'off') {
+            // Dynamically construct the message based on conflicting ports in $overlap
+            $portOverlap = implode(', ', $overlap);
+            $portOverlapMessage = [];
+            $portOverlapMessageConstruction = implode(' and ', $portOverlapMessage);
```

Comment:
doesn't play nice with translations. personally I would opt for a smaller, condensed, message explaining the options in play here, but if you do want to construct the message,  just push `$portOverlap ` in the message and remove the line above.

Reply:
I have removed the message construct in favor of a condensed message.